File: MasterErp.Entities/Common/ItemModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MasterErp.Entities.Common
{
    public class ItemModel
    {
        public int ItemId { get; set; }
        public string ItemNameAr { get; set; }
        public string ItemNameEn { get; set; }
        public int UnitId { get; set; }
        public string UnitNameAr { get; set; }
        public string UnitNameEn { get; set; }
        public double Price { get; set; }
        public double Quantity { get; set; }
        public double TotalValue { get; set; }
        public bool? IsActive { get; set; }
        public string InsertUser { get; set; }
        public DateTime? InsertDate { get; set; }

    }
}

File: RetailStation.Angular/src/app/components/Shared/models/ItemModel.ts
import { CreatorModel } from "src/app/components/Shared/models/CreatorModel";

export interface ItemModel extends CreatorModel {
    itemId?: number | null;
    nameAR: string;
    nameEN: string;
    unitId: number | null;
    unitName: string;
    purchaseUnitId: number | null;
    purchaseUnitName: string;
    itemCategoryId: number | null;
    itemCategoryName: string;
    cost: number | null;
    purchasePrice: number | null;
    yield: number | null;
    convertRatio: number | null;
    isActive: boolean | null;
    itemTypeId: number | null;
    supplierId: number | null;
    supplierName: string;
    disabled: boolean | null;
    image: File | null;
    imageUrl: string;

}


export class OrderDetailModel {
    itemId: number;
    itemNameAr: string;
    itemNameEn: string;
    unitId: number;
    unitNameAr: string;
    unitNameEn: string;
    price: number;
    quantity: number;
    itemTotalValue: number;
}
